Amazon.IonObjectMapper/IonObjectSerializer.cs

```diff
@@ -25,6 +25,19 @@ public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions
 
         public override object Deserialize(IIonReader reader)
         {
+            var ionConstructors = targetType.GetConstructors(BINDINGS).Where(IsIonConstructor).Take(2);
+            if (ionConstructors.Any())
+            {
+                if (ionConstructors.Count() > 1)
+                {
+                    throw new InvalidOperationException(
+                        $"Only one constructor in class {targetType.Name} may be annotated " +
+                        "with the [IonConstructor] attribute and more than one was detected");
+                }
+
+                return this.DeserializeWithIonConstructor(ionConstructors.First(), reader);
```

Comment:
The only thing that is special about `IonConstructor` is how we create an instance of the object. Everything else is the same. Therefore it's just line 47 that you need to do this special constructor thing on.

Reply:
The two deserialization implementations are now merged into a single implementation.